Sources_v3/StreamChatUI/ChatChannelList/ChatSwipeableListItemView.swift

```diff
@@ -0,0 +1,101 @@
+    public private(set) lazy var deleteButton: UIButton = UIButton().withoutAutoresizingMaskConstraints
+    public private(set) lazy var actionButtonStack: UIStackView = UIStackView().withoutAutoresizingMaskConstraints
+
+    // MARK: - View
+
+    override public func setUpLayout() {
+        super.setUpLayout()
+        addSubview(cellContentView)
+        cellContentView.pin(anchors: [.top, .bottom, .width], to: self)
+        addSubview(actionButtonStack)
+        actionButtonStack.addArrangedSubview(deleteButton)
+
+        actionButtonStack.pin(anchors: [.top, .bottom], to: self)
+
+        actionButtonStack.axis = .horizontal
+        actionButtonStack.alignment = .fill
+
+        deleteButton.widthAnchor.constraint(equalTo: deleteButton.heightAnchor).isActive = true
+        cellContentView.trailingAnchor.constraint(equalTo: actionButtonStack.leadingAnchor).isActive = true
+        trailingConstraint = trailingAnchor.constraint(equalTo: cellContentView.trailingAnchor)
+        trailingConstraint?.isActive = true
+    }
+
+    override public func setUp() {
+        super.setUp()
+
+        deleteButton.addTarget(self, action: #selector(didTapDelete), for: .touchUpInside)
+
+        let pan = UIPanGestureRecognizer(target: self, action: #selector(didPan(_:)))
+        pan.delegate = self
+        addGestureRecognizer(pan)
+    }
+
+    override public func defaultAppearance() {
+        super.defaultAppearance()
+        deleteButton.setImage(UIImage(named: "icn_delete", in: .streamChatUI), for: .normal)
+        deleteButton.backgroundColor = uiConfig.colorPalette.channelListActionsBackgroundColor
+        deleteButton.tintColor = uiConfig.colorPalette.channelListActionDeleteChannel
+    }
+
+    // MARK: - Button actions
+
+    @objc func didTapDelete() {
+        deleteButtonAction?()
+    }
+
+    // MARK: Gesture recognizer
+
+    public func gestureRecognizer(
+        _ gestureRecognizer: UIGestureRecognizer,
+        shouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer
+    ) -> Bool {
+        true
+    }
+
+    @objc func didPan(_ gesture: UIPanGestureRecognizer) {
+        if gesture.state == .began {
+            startedValue = trailingConstraint?.constant ?? 0
+        }
+
+        let translation = gesture.translation(in: self)
+        let maxValue = actionButtonStack.frame.width
+        let value: CGFloat
+        if translation.x < 0 {
+            value = min(-translation.x, maxValue)
+        } else {
+            value = max(0, startedValue - translation.x)
+        }
+
+        if gesture.state == .cancelled || gesture.state == .failed {
+            trailingConstraint?.constant = 0
+            return
+        }
+        if gesture.state == .ended {
+            if value > maxValue / 2 {
+                trailingConstraint?.constant = maxValue
+            } else {
+                trailingConstraint?.constant = 0
+            }
+            return
+        }
```

Comment:
It would be nice to have these changes animated.

Reply:
Added simple `UIView` animation for `layoutIfNeeded()`. Looks bit more nice.